eval/src/vespa/eval/eval/test/reference_operations.cpp

```diff
@@ -283,5 +283,32 @@ TensorSpec ReferenceOperations::rename(const TensorSpec &a, const std::vector<ve
     return result;
 }
 
+TensorSpec ReferenceOperations::lambda(const vespalib::string &type_in, lambda_fun_t fun) {
+    ValueType type = ValueType::from_spec(type_in);
+    TensorSpec result(type.to_spec());
+    if (type.is_error()) {
+        return result;
+    }
+    std::vector<size_t> sizes;
```

Comment:
adding `const auto & dimensions = type.dimensions()` here would improve readability;
note there's no need for the `sizes` vector, you can just use dimensions[idx].size directly.